### PR TITLE
Feature/2284 - Delete Item endpoint and unit tests

### PR DESCRIPTION
**Pull request by Akrish17**
[2284 - Delete Item endpoint and unit tests]

#### 📲 What

Implemented new endpoint to delete an item from the menu

#### 🤔 Why

Can only delete an item from the menu using this endpoint

#### 🛠 How

Pass the  menuId, categoryId and itemId  to be deleted in the request

#### 👀 Evidence

Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR
image

#### 🕵️ How to test

Notes for QA
Swagger JSON is available here: http://localhost:9000/swagger/oas.json
UI Swagger here: http://localhost:9000/swagger/index.html

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/menu/api/v1/impl/DeleteItemControllerImpl.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  private DeleteItemHandler handler;
+
+  public DeleteItemControllerImpl(DeleteItemHandler handler) {
+    this.handler = handler;
+  }
+
+  @Override
+  public ResponseEntity<Void> deleteItem(UUID menuId, UUID categoryId,
+                                         UUID itemId, String correlationId) {
+    DeleteItemCommand command = map(correlationId, menuId, categoryId, itemId);
```
Comment (peet-j) [inline java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/menu/api/v1/impl/DeleteItemControllerImpl.java:29]: again

File: java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/menu/api/v1/impl/QueryMenuControllerImpl.java [hunk 2]
```diff
@@ -33,7 +33,8 @@
 
     private MenuQueryService menuQueryService;
 
-    public QueryMenuControllerImpl(DomainToDtoMapper mapper, MenuQueryService menuQueryService) {
```
Comment (ekartsonakisamido) [inline java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/menu/api/v1/impl/QueryMenuControllerImpl.java:36]: Can the `MenuQueryService` be injected?

File: java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/menu/domain/Category.java [hunk 3]
```diff
@@ -15,5 +18,20 @@
 
   private String description;
 
-  private List<Item> items;
+  @Builder.Default
+  private List<Item> items = new ArrayList<>();
+
+  public Category addUpdateItem(Item item) {
```
Comment (ekartsonakisamido) [inline java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/menu/domain/Category.java:24]: It's not clear to me what this method's doing. Do we follow a naming convention?

**Review by ekartsonakisamido — COMMENTED** [review]
Looks good, but I've a few general architecture questions. Can we have a call on Friday to go through these?

File: java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/menu/api/v1/impl/QueryMenuControllerImpl.java (hunk 2 above)
Comment (Akrish17) [reply java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/menu/api/v1/impl/QueryMenuControllerImpl.java:36]: the common practice now is using constructor injection.

**Pull request merged**